### PR TITLE
thefuck 3.4

### PR DESCRIPTION
**Pull request by scorphus**
Hello, dear Homebrew maintainers!

Here we go again, updating The Fuck from 3.3 to 3.4! This new version brings a lot of improvements, new features and bug fixes. All in all it is a bit faster and more resilient. Yay!

Thanks!

### CODE REVIEW

**Comment by scorphus** [comment]
Looks like it's not only `pip` which is missing, I'm experiencing another problem with `gdm`.

I will fix these issues and force-push to this PR.

**Comment by scorphus** [comment]
There might be an issue with `gdbm` in cases where the user had interchanged between the **system's** and **virtualenv's** Python executable. I'll fix this upstream and then add a patch to this formula. Please hang on :smiley:

**Comment by scorphus** [comment]
YAY! We're good to go! 2b10161 has the patch!

A very few users would be affected by what this patch fixes, though... but we're better safe than sorry :smile: 

Upstream will soon be patched, too: nvbn/thefuck#473

File: Library/Formula/thefuck.rb [hunk 1]
```diff
@@ -45,10 +45,18 @@ class Thefuck < Formula
     sha256 "1a089279d5de2471c47624d4463f2e5b3fc6a2cf65045c39bf714fc461a25206"
   end
 
+  resource "pip" do
+    url "https://pypi.python.org/packages/source/p/pip/pip-8.0.3.tar.gz"
+    sha256 "30f98b66f3fe1069c529a491597d34a1c224a68640c82caf2ade5f88aa1405e8"
+  end
+
+  # Temporarily fix cache incompatibilities issues. FIXME: Remove this in 3.5!
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/thefuck.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (scorphus) [inline Library/Formula/thefuck.rb]: Absolutely! :+1: Will do

**Comment by scorphus** [comment]
@mikemcquaid Done so, please review!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by scorphus** [comment]
I'm glad to contribute to such an awesome project! :heart: :beer:

**Pull request closed without merging**